unit-tests/post-processing/test-post-processing-filters.py

```diff
@@ -0,0 +1,469 @@
+    depth_intrinsics.coeffs = [0, 0, 0, 0, 0]
+    return depth_intrinsics
+
+def create_video_stream(test_cfg, depth_intrinsics):
+    vs = rs.video_stream()
+    vs.type = rs.stream.depth
+    vs.index = 0
+    vs.uid = 0
+    vs.width = test_cfg.input_res_x
+    vs.height = test_cfg.input_res_y
+    vs.fps = 30
+    vs.bpp = 2
+    vs.fmt = rs.format.z16
+    vs.intrinsics = depth_intrinsics
+    return vs
+
+def create_frame(test_cfg, depth_stream_profile, index):
+    frame = rs.software_video_frame()
+    frame.pixels = test_cfg._input_frames[index]
+    frame.bpp = 2
+    frame.stride = test_cfg.input_res_x * 2
+    frame.timestamp = index + 1
+    frame.domain = rs.timestamp_domain.system_time
+    frame.frame_number = index
+    frame.profile = depth_stream_profile.as_video_stream_profile()
+    frame.depth_units = test_cfg.depth_units
+    return frame
+
+
+################################################################################################
+with test.closure("Post-Processing Filters sequence validation"):
+    test_cfg = ppf_test_config()
+    for first_element, second_element in ppf_test_cases:
+        # Load the data from configuration and raw frame files
+        if not load_test_configuration(first_element, test_cfg):
+            continue
+
+        ppf = post_proccesing_filters()
+        ppf.configure(test_cfg)
+
+        sw_dev = rs.software_device()
+        depth_sensor = sw_dev.add_sensor("Depth")
+
+        depth_intrinsics = create_depth_intrinsics(test_cfg)
+
+        vs = create_video_stream(test_cfg, depth_intrinsics)
+        depth_stream_profile = depth_sensor.add_video_stream(vs)
+        depth_sensor.add_read_only_option(rs.option.depth_units, test_cfg.depth_units)
+        depth_sensor.add_read_only_option(rs.option.stereo_baseline, test_cfg.stereo_baseline_mm)
+
+        sw_dev.create_matcher(rs.matchers.dlr_c)
+        sync = rs.syncer()
+
+        depth_sensor.open(depth_stream_profile)
+        depth_sensor.start(sync)
+
+        frames = test_cfg.frames_sequence_size if test_cfg.frames_sequence_size > 1 else 1
+        for i in range(frames):
+            frame = create_frame(test_cfg, depth_stream_profile, i)
+            depth_sensor.on_video_frame(frame)
+
+            fset = sync.wait_for_frames()
+            depth = fset.first_or_default(rs.stream.depth)
+
+            #the actual filters are being applied
+            filtered_depth = ppf.process(depth)
+            validate_ppf_results(filtered_depth, test_cfg, i)
+
+        depth_sensor.stop()
+        depth_sensor.close()
+################################################################################################
+with test.closure("Post-Processing Filters metadata validation"):
+    test_cfg = ppf_test_config()
+    for first_element, second_element in ppf_test_cases:
+        # Load the data from configuration and raw frame files
+        if not load_test_configuration(first_element, test_cfg):
```

Comment:
How long does it take for the test to run?
Since it loads from disk for both test cases, and actually they do mostly the same, maybe we can do the shared part together.

Reply:
We can do it together. The legacy tests were separated so I did here the same.

Reply:
OK. I agree that separated test cases is a better consept. Just wondered how much time does it costs us here.
Maybe it is a long test and we can cut the time by half..

Reply:
Each test doesn't take long so I think we can leave it like that.